plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoIcebergRestCatalogFactory.java

```diff
@@ -56,6 +60,10 @@ public class TrinoIcebergRestCatalogFactory
     private final SecurityProperties securityProperties;
     private final boolean uniqueTableLocation;
     private final TypeManager typeManager;
+    private final boolean caseInsensitiveNameMatching;
+    private final Cache<Namespace, Namespace> remoteNamespaceMappingCache;
+    private final Cache<TableIdentifier, TableIdentifier> remoteTableMappingCache;
```

Comment:
`remoteTableMappingCache` looks having a cache of views. Could you confirm that? You can check by creating a view, setting the breakpoint `listTables` and executing `SHOW TABLES` statement.

Reply:
Nice find 🙌 , thanks for debugging it. Apparently some methods require mapped object names without it being cached. With https://github.com/trinodb/trino/pull/23867/commits/23f0b4b8facd1cdf0323e3abf3f7d4a007fda560, now table/view cache has correct entries. I'll explore if there are possibility of sharing table/view cache now given methods interfering caches now uses live mapped entries instead of loading other object's cache.

Reply:
Added a commit to use single cache for table/view. Please review.